lucene/suggest/src/java/org/apache/lucene/search/spell/DirectSpellChecker.java

```diff
@@ -195,9 +197,27 @@ public int getMinQueryLength() {
    * metric.
    */
   public void setMinQueryLength(int minQueryLength) {
+    if (minQueryLength > this.maxQueryLength)
```

Comment:
Actually I think we don't need these checks because whatever the value for minQueryLength and maxQueryLength, the logic is always the same, no surprise.
So personally I would remove these checks, but I'm ok to keep them, your choice.